### PR TITLE
call MPI_Init_thread when necessary

### PR DESCRIPTION
**Pull request by bryevdv**
Suggestion from @eddy16112 re: different MPI setup on some specific systems.

### CODE REVIEW

File: src/core/comm/mpi_comm.cc [hunk 1]
```diff
@@ -69,10 +69,8 @@ MPINetwork::MPINetwork(int argc, char* argv[])
   int provided, init_flag = 0;
   CHECK_MPI(MPI_Initialized(&init_flag));
   if (!init_flag) {
-    log_coll.fatal(
-      "MPI has not been initialized, it should be initialized by "
-      "the networking backend.");
-    LEGATE_ABORT;
+    MPI_Init_thread(0, 0, MPI_THREAD_MULTIPLE, &provided);
+    self_mpi_init = true;
```
Comment (manopapad) [inline src/core/comm/mpi_comm.cc:73]:
```suggestion
    self_init_mpi = true;
```

Comment (bryevdv) [reply src/core/comm/mpi_comm.cc:73]: :exclamation: why did this compile?? (I did compile it locally)

File: src/core/comm/mpi_comm.cc [hunk 2]
```diff
@@ -69,10 +69,8 @@ MPINetwork::MPINetwork(int argc, char* argv[])
   int provided, init_flag = 0;
   CHECK_MPI(MPI_Initialized(&init_flag));
   if (!init_flag) {
-    log_coll.fatal(
-      "MPI has not been initialized, it should be initialized by "
-      "the networking backend.");
-    LEGATE_ABORT;
+    MPI_Init_thread(0, 0, MPI_THREAD_MULTIPLE, &provided);
```
Comment (eddy16112) [inline src/core/comm/mpi_comm.cc:72]: Maybe add a `log_coll.info(...)` to mention MPI is initialized by legate. Mainly for debugging. 
Could you please also update line 107 to use the log_coll.info. Sorry, it was a mistake of mine.

**Review by eddy16112 — COMMENTED** [review]
LGTM, just let a comment about debug print.

File: src/core/comm/mpi_comm.cc (hunk 2 above)
Comment (bryevdv) [reply src/core/comm/mpi_comm.cc]: https://github.com/nv-legate/legate.core/pull/711/commits/6b8a8c3d11d4d7cf3c0bcd6df378aaef39a35231

**Review by eddy16112 — APPROVED** [review]

**Pull request merged**